File: src/main/java/basics/InputMethod.java
package basics;

import java.util.Scanner;

public class InputMethod {

    public static void main(String[] args) {
        System.out.println("Enter first number");
        int a =getIntFromScanner();
        System.out.println("Enter second number");
        int b = getIntFromScanner();
        System.out.println("Sum is "+(a+b));
    }

    public static int getIntFromScanner(){
        Scanner scanner = new Scanner(System.in);
        boolean isValidData = false;
        int val=0;
        do{
            try{
                System.out.println("Enter a Number");
                String num = scanner.nextLine();
                val= Integer.parseInt(num);
                isValidData=true;
            } catch (NumberFormatException e) {
                System.out.println("Invalid Data, Try Again");
            }
        } while(!isValidData);
        return val;

    }
}

File: python/basics/input_method.py
class InputMethod:
    @staticmethod
    def main():
        print("Enter first number")
        a = InputMethod.get_int_from_input()
        print("Enter second number")
        b = InputMethod.get_int_from_input()
        print(f"Sum is {a + b}")

    @staticmethod
    def get_int_from_input():
        is_valid_data = False
        val = 0

        while not is_valid_data:
            try:
                print("Enter a Number")
                num = input()
                val = int(num)
                is_valid_data = True
            except ValueError:
                print("Invalid Data, Try Again")

        return val


if __name__ == "__main__":
    InputMethod.main()
